### PR TITLE
Dev branch unit tests

### PR DESCRIPTION
**Pull request by VenelinKadankov**

### CODE REVIEW

**Review by Tsa-chi — COMMENTED** [review]
LGTM

File: UnitTests/Tests/UnitTest1.cs [hunk 1]
```diff
@@ -0,0 +1,144 @@
+namespace Tests;
+
+using System.Numerics;
+using UTest_NUnit;
+
+public class Tests
```
Comment (YuliaStarostenko) [inline UnitTests/Tests/UnitTest1.cs]: suggest to change the Class name to more describing one(not just Test)

File: UnitTests/Tests/UnitTest1.cs [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+{
+
+    [SetUp]
+    public void Setup()
```
Comment (YuliaStarostenko) [inline UnitTests/Tests/UnitTest1.cs]: As Set up is empty, it is better to delete the method

File: UnitTests/Tests/UnitTest1.cs [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    {
+    }
+
+    [Test]
+    public void TestSumWithValidData()
```
Comment (YuliaStarostenko) [inline UnitTests/Tests/UnitTest1.cs]: Suggest to change the name to TestSumWithPositiveNumbers (like you have below for Negative) as just Valid data is too general(e.i testing with negative numbers is still testing with valid data)

File: UnitTests/Tests/UnitTest1.cs [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    public void TestSumWithNegativeNumbers()
+    {
+        var actual = Sumator.Sum(new int[] { -6, -10 });
+        BigInteger expected = -16;
+
+        Assert.That(actual, Is.EqualTo(expected));
+    }
+
+    [Test]
+    public void TestSumWithbigNumbers()
```
Comment (YuliaStarostenko) [inline UnitTests/Tests/UnitTest1.cs]: ...BigNumbers

File: UnitTests/Tests/UnitTest1.cs [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    {
+        var actual = Sumator.Sum(new int[] { 2000000000, 2000000000 });
+        BigInteger expected = 4000000000;
+
+        Assert.That(actual, Is.EqualTo(expected));
+    }
+
+    [Test]
+    public void TestSumWithPositiveNegativeNumbers()
```
Comment (YuliaStarostenko) [inline UnitTests/Tests/UnitTest1.cs]: TestSumWithPositiveAndNegativeNumbers(with And is more readable)

File: UnitTests/Tests/UnitTest1.cs [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+    [Test]
+    public void TestAverageEmptyArrThrows()
+    {
+        var actual = () => Sumator.Average(Array.Empty<int>());
+
+        Assert.That(actual, Throws.InstanceOf<IndexOutOfRangeException>());
+    }
+
+    [Test]
+    public void TestAverageOneElementArray()
```
Comment (YuliaStarostenko) [inline UnitTests/Tests/UnitTest1.cs]: TestAverageWithOneElementArray

File: UnitTests/Tests/UnitTest1.cs [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+    public void TestAverageValidNumbers()
+    {
+        var actual = Sumator.Average(new int[] { 200, 100 });
+        var expected = 150;
+
+        Assert.That(actual, Is.EqualTo(expected));
+    }
+
+    [Test]
+    public void TestAverageValidNumbersNotIntResult()
```
Comment (YuliaStarostenko) [inline UnitTests/Tests/UnitTest1.cs]: Suggest to reconsider the test name : after TestAverage... the rest is not undersandable(not understandable what you test here)

File: UnitTests/Tests/UnitTest1.cs [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+    {
+        var actual = Sumator.Average(new int[] { 200, 101 });
+        var expected = 150.5;
+
+        Assert.That(actual, Is.EqualTo(expected));
+    }
+
+    [Test]
+    public void TestAverageEmptyArrThrows()
```
Comment (YuliaStarostenko) [inline UnitTests/Tests/UnitTest1.cs]: Better to refactor the name: smth like: TestAverageThrowsExeption_When_TesstedWithEmptyArray

File: UnitTests/Tests/UnitTest1.cs [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+    public void TestAverageOneElementArray()
+    {
+        var actual = Sumator.Average(new int[] { 10 });
+        var expected = 10;
+
+        Assert.That(actual, Is.EqualTo(expected));
+    }
+
+    [Test]
+    public void TestAverageNegativeNumbers()
```
Comment (YuliaStarostenko) [inline UnitTests/Tests/UnitTest1.cs]: TestAverageWithNegativeNumbers

File: UnitTests/Tests/UnitTest1.cs [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+    {
+        var actual = Sumator.Average(new int[] { -10, -20, -30 });
+        var expected = -20;
+
+        Assert.That(actual, Is.EqualTo(expected));
+    }
+
+    [Test]
+    public void TestAverageNegativeAnPositiveNumbers()
```
Comment (YuliaStarostenko) [inline UnitTests/Tests/UnitTest1.cs]: TestAverageNegativeAndPositiveNumbers(typo :))

File: UnitTests/Tests/UnitTest1.cs [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+    public void TestMinimumValidData()
+    {
+        var actual = Sumator.MinimumValue(new int[] { 3, 5, -100, 99 });
+        var expected = -100;
+
+        Assert.That(actual, Is.EqualTo(expected));
+    }
+
+    [Test]
+    public void TestMinimumEmptyData()
```
Comment (YuliaStarostenko) [inline UnitTests/Tests/UnitTest1.cs]: ...EmptyArray

File: UnitTests/Tests/UnitTest1.cs [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+    [Test]
+    public void TestMinimumNullData()
+    {
+        var actual = () => Sumator.MinimumValue(null);
+
+        Assert.That(actual, Throws.InstanceOf<ArgumentNullException>());
+    }
+
+    [Test]
+    public void TestMinimumSimilarData()
```
Comment (YuliaStarostenko) [inline UnitTests/Tests/UnitTest1.cs]: TestMinimumWithEqualNumbers

**Pull request merged**